File: services/gateway/internal/dto/transaction.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type TransferRequest struct {
	FromAccountId  uuid.UUID `json:"from_account_id"`
	ToAccountId    uuid.UUID `json:"to_account_id"`
	Amount         int64     `json:"amount"`
	Currency       string    `json:"currency"`
	IdempotencyKey string    `json:"idempotency_key"`
}

type TransactionResponse struct {
	TransactionId     uuid.UUID `json:"transaction_id"`
	InitiatorId       uuid.UUID `json:"initiator_id"`
	FromAccountId     uuid.UUID `json:"from_account_id"`
	ToAccountId       uuid.UUID `json:"to_account_id"`
	Amount            int64     `json:"amount"`
	Currency          string    `json:"currency"`
	IdempotencyKey    string    `json:"idempotency_key"`
	TransactionStatus string    `json:"transaction_status"`
	CreatedAt         time.Time `json:"created_at"`
}

File: frontend/src/models/transaction.ts
export interface Transaction {
    transaction_id: string
    initiator_id: string
    from_account_id: string
    to_account_id: string
    amount: number
    currency: string
    idempotency_key: string
    transaction_status: string
    created_at: string
}

export interface TransferRequest {
    from_account_id: string
    to_account_id: string
    amount: number
    currency: string
    idempotency_key: string
}